File: app/src/main/java/com/dhsdevelopments/potato/clientapi/message/MessageElementCodeBlock.java
package com.dhsdevelopments.potato.clientapi.message;

public class MessageElementCodeBlock extends MessageElement
{
    private final String language;
    private final String code;

    public MessageElementCodeBlock( String language, String code ) {
        this.language = language;
        this.code = code;
    }

    @Override
    public CharSequence getSpannable() {
        return code;
    }

    @Override
    public String toString() {
        return "MessageElementCodeBlock[" +
                       "language='" + language + '\'' +
                       ", code='" + code + '\'' +
                       "] " + super.toString();
    }
}

File: app/src/main/java/com/dhsdevelopments/potato/clientapi/message/MessageElementCodeBlock.kt
package com.dhsdevelopments.potato.clientapi.message

class MessageElementCodeBlock(private val language: String, private val code: String) : MessageElement() {

    override fun makeSpan(): CharSequence {
        return code
    }

    override fun toString(): String {
        return "MessageElementCodeBlock[" +
                "language='" + language + '\'' +
                ", code='" + code + '\'' +
                "] " + super.toString()
    }
}
